sys/evtimer/evtimer.c

```diff
@@ -100,7 +100,12 @@ static void _set_timer(evtimer_t *evtimer)
     DEBUG("evtimer: now=%" PRIu32 " ms setting ztimer to %" PRIu32 " ms\n",
           evtimer->base, next_event->offset);
 
-    ztimer_set(ZTIMER_MSEC, &evtimer->timer, next_event->offset);
+    if(next_event->offset==0) {
```

Comment:
I don't quite understand this change, this callback will not be ran in ISR context

Reply:
as it happens with xtimer (shot timeout-> xtimer_spin -> xtimer_spin somehow circles around a scheduling issue) (this is the reason the test is not failing on native with xtimer)

Reply:
talking about this test

https://github.com/RIOT-OS/RIOT/pull/17357#issuecomment-988817899

> `evtimer_underflow` is failing with `USEMODULE=evtimer_on_ztimer` before and after this commit so there is a bug that this does not yet fix

Reply:
this should only be an issue on native (afaik) though

Reply:
maybe emulating that bad behavior of xtimer isnt the best way of solving for the test